packages/design/src/DataTable/useTable.ts

```diff
@@ -172,14 +173,39 @@ function searchAndFilterCb<T>(
   searchValue: string,
   propName: keyof T & string
 ) {
-  if (propName === 'tags') {
-    return targetValue.some(item => {
-      return item.toLocaleLowerCase().includes(searchValue.toLocaleLowerCase());
-    });
-  }
   if (propName.toLocaleLowerCase().includes('date')) {
     return displayDate(targetValue).includes(searchValue);
   }
+  if (propName.toLocaleLowerCase().includes('time')) {
+    return displayDateTime(targetValue).includes(searchValue);
+  }
+  if (typeof targetValue === 'boolean') {
+    return (
+      propName.toLocaleLowerCase().includes(searchValue.toLocaleLowerCase()) &&
+      targetValue
+    );
+  }
+
+  // For searching through elements of an array
+  if (Array.isArray(targetValue)) {
+    return targetValue.some(item => {
+      if (typeof item === 'object') {
+        for (const key in item) {
+          if (
+            item[key]
+              .toLocaleLowerCase()
+              .includes(searchValue.toLocaleLowerCase())
+          ) {
+            return true;
+          }
+        }
+      } else {
+        return item
```

Comment:
This assumes it's a `string`, it appears that it could be possible that it's not as you're only validating the type of the value if it's _not_ an `array`. You might want to combine these checks so that you're not handling array values differently. You could do this by simply placing the `targetValue` into an array if it isn't one and then looping over each value with the same validation checks.